packages/core/src/options.ts

```diff
@@ -124,11 +152,13 @@ export default interface Options {
       };
 }
 
-export function optionsWithDefaults(options?: Options): Required<Options> {
+export type CoreOptionsType = Required<
+  Omit<Options, 'staticHelpers' | 'staticModifiers' | 'staticComponents' | 'staticInvokables'>
+> &
+  Pick<Options, 'staticHelpers' | 'staticModifiers' | 'staticComponents' | 'staticInvokables'>;
+
+export function optionsWithDefaults(options?: Options): CoreOptionsType {
   let defaults = {
-    staticHelpers: false,
-    staticModifiers: false,
-    staticComponents: false,
```

Comment:
shouldn't this add staticInvokables default?

Reply:
so the defaults are there to make sure the types are always right. Essentially we can't set a default for staticInvokables because we would need to set it only if you haven't defined the other settings. I asked Ed about it and he just said to make everything optional and we can rationalise it in the constructor 👍